File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        n = len(heights)-1
        i,j=0,n
        maxArea =0
        while i<j :
            maxArea = max(min(heights[i],heights[j])*(j-i),maxArea)
            if heights[i]<heights[j]:
                i+=1
            else:
                j-=1
        return maxArea            

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        Set<Character> set = new HashSet<>();
        int max =0;
        int left =0;
        for(char c : s.toCharArray()){
            if(set.contains(c)){
                while(set.contains(c)){
                set.remove(s.charAt(left));
                left++;
                }
            }
            set.add(c);
            max = Math.max(set.size(),max);
        }
        return max;
    }
}
